test/messingaround/messingaround/MainActivity.kt

```diff
@@ -1,56 +1,47 @@
 package com.test.messingaround.messingaround
 
-import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
 import android.util.Log
 import io.reactivex.Observable
 import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.rxkotlin.Observables
 import io.reactivex.schedulers.Schedulers
 import kotlinx.android.synthetic.main.activity_main.*
+import java.util.concurrent.TimeUnit
 
 class MainActivity : AppCompatActivity() {
+    private var stringsDisposable: Disposable? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        // Async method. Will stall thread it runs on until true.
-        fun isHalfDone() : Boolean {
-            while (mainText.rotation != 180f) {
-                Thread.sleep(100) // Wait 100ms before checking again, even though on its own thread.
-            }
-            return true
-        }
+        // On click listener for testButton.
+        testButton.setOnClickListener {
+            val userString = inputText.text
 
-        var state = 0 // Button output has 2 states. Just for testing stuff.
+            val delayedStrings = Observables
```

Comment:
Instead of starting the animation in two different places, we can just emit two events. Each of them will trigger the animation. Try reading the docs for the methods I used here, and ask me if something is unclear.

Also note that the function used here is from `RxKotlin` library which can be included as follows:
```
implementation 'io.reactivex.rxjava2:rxkotlin:2.3.0'
```